pkg/cloud/services/eks/cluster.go

```diff
@@ -550,6 +573,29 @@ func (s *Service) describeEKSCluster(eksClusterName string) (*eks.Cluster, error
 	return out.Cluster, nil
 }
 
+func (s *Service) updateEncryptionConfig(updatedEncryptionConfigs []*eks.EncryptionConfig) error {
+	input := &eks.AssociateEncryptionConfigInput{
+		ClusterName:      &s.scope.Cluster.ClusterName,
```

Comment:
You will need to use `s.scope.KubernetesClusterName()` instead here just in case someone has given an explicit EKS cluster name in the AWSManagedControlPlane.